src/image/p_grayscale.c

```diff
@@ -0,0 +1,35 @@
+#include <pal.h>
+
+/**
+ * RGB to Grayscale converter
+ *
+ * Each channel is replaced with the grayscale value.
+ * The grayscale value is computed as 0.299*R + 0.587*G + 0.114*B
```

Comment:
Might be an idea to give the coefficients as an argument to the function. The current static coeficients are for BT.601, HDTV ( BT. 709) has them slightly different.